Add console command aliases

a3020 opened the pull request:
I think it makes more sense to type `c5:translate-package`, than to type `c5:package-translate`.

Because the commands are alphabatically sorted, I added the following aliases:

- `c5:install-language`
- `c5:install-package`
- `c5:pack-package`
- `c5:translate-package`
- `c5:uninstall-package`
- `c5:update-package`

<img width="444" alt="aliases" src="https://user-images.githubusercontent.com/1431100/31587071-24a1de26-b1db-11e7-871b-178273ce5565.png">

mlocati commented:
What about changing the names to `c5:package:translate` (and so on), and set the old names ad aliases?

a3020 commented:
I can change it, but I thought the current commands were on purpose: to group the commands. Therefore I created aliases.

mlocati commented:
> I can change it, but I thought the current commands were on purpose: to group the commands

That's exactly why I gave them those names.
But using the `c5:package:...` notation will be handy if someone wants to list all the package-related commands: it'd enough to write `concrete/bin/concrete5 list c5:package` to get
```
  c5:package:install    Install a concrete5 package
  c5:package:pack       Process a package (expand PHP short tags, compile icons and translations, create zip archive)
  c5:package:translate  Creates or updates translations of a concrete5 package
  c5:package:uninstall  Uninstall a concrete5 package
  c5:package:update     Update a concrete5 package
```

a3020 commented:
Updated PR:

`$ concrete/bin/concrete5`
<img width="615" alt="c5" src="https://user-images.githubusercontent.com/1431100/31588451-1b28e7d8-b1f2-11e7-9e7d-400d56b3af53.png">

`$ concrete/bin/concrete5 package:list`
<img width="657" alt="c5-package" src="https://user-images.githubusercontent.com/1431100/31588469-33e800c4-b1f2-11e7-9ef5-814dd76889d4.png">

@mlocati I understand now what you mean. I've updated the PR. Does it make more sense to you now?

mlocati commented:
Yes! About the error in the second screenshot: try
```
concrete/bin/concrete5 list c5:package
```

And I'm not 100% sure about introducing new aliases like `c5:install-package`: I can't see why they could be useful...

a3020 commented:
> And I'm not 100% sure about introducing new aliases like c5:install-package: I can't see why they could be useful...

I think it reads more like a normal sentence: 'install a package' => `install-package`. I also think it fits better looking at the other commands, e.g. it currently is `c5:compare-schema` instead of `c5:schema-compare`.

I don't have any problems changing the commit with just the `:` syntax.

mlocati commented:
> I think it reads more like a normal sentence: 'install a package' => install-package

IMHO this isn't an argument strong enough to add command aliases... But obviously that's just a personal opinion.

> I also think it fits better looking at the other commands, e.g. it currently is c5:compare-schema instead of c5:schema-compare.

IMHO, having a "namespaced" set of commands is nicer:
- do you want to list the concrete5-specific commands? Use `list c5`
- do you want to list the package-specific commands? Use `list c5:package`

KorvinSzanto commented on concrete/src/Console/Command/InstallLanguageCommand.php:
```diff
@@ -21,6 +21,7 @@ protected function configure()
         $errExitCode = static::RETURN_CODE_ON_FAILURE;
         $this
         ->setName('c5:language-install')
```
This still needs updating, no?

KorvinSzanto commented:
Why do we even have that c5 namespace? All the commands in the console are there to effect c5 so it really doesn't make sense to include that namespace. 

IMO These commands should be:

```php
concrete5 package:install package_handle
concrete5 package:uninstall package_handle
```

not `concrete5 c5:package:install package_handle`

mlocati commented:
> IMO These commands should be:
> 
> ```
> concrete5 package:install package_handle
> concrete5 package:uninstall package_handle
> ```
> not `concrete5 c5:package:install package_handle`

I'd prefer to keep the `c5` namespace for all the core commands, in order to avoid namespace clashes and having a simple way to list the core commands (`list c5`)

KorvinSzanto commented:
It just feels super redundant to me. Can you describe a situation where a namespace collision would occur?

As for the list command, does `list c5` work if only the aliases start with `c5:`?

mlocati commented:
> Can you describe a situation where a namespace collision would occur?

When a 3rd party package defines commands with the `package` namespace. I like to think the `c5` as the core commands namespace (exactly like the `Concrete\Core` is the core namespace for PHP code)

> As for the list command, does list c5 work if only the aliases start with c5:?

I just tested it: yes

aembler commented:
Agree with @mlocati on this - we have used custom name spaces for package cli scripts and having the differentiation is nice. Now I might be persuaded to change it to “core” from “c5” since korvin is right - but there are a few third party cli tools in these for doctrine and doctrine migrations so leaving the namespace unchanged and in place might be the best option

KorvinSzanto commented:
Fair enough, this looks good to me! :+1:

mlocati commented:
PS: correctly namespaced commands are very handy... Instead of running
```
concrete5 c5:package:uninstall <package_handle>
```

you can specify parts of the namespace/command, like for example
```
concrete5 c5:pack:un <package_handle>
```
or even
```
concrete5 c5:p:un <package_handle>
```

:wink:
